### PR TITLE
Update GraphQL Server to using apollo server

### PR DESCRIPTION
**Pull request by nattiechan**
This is a pre-cursor to implementing the GraphQL mock server using apollo server.

`apollo-server-express` is scheduled to be deprecated at the end of 2023. This commit updates the GraphQL server to use `@apollo/server` instead.

Note that Apollo server 4 requires top-level async/await for the server to start properly, which can only be done in a module. Since most backend logic in Swell uses CommonJS, we need to change this file to `.mjs` to make things work. The downside to making this as a `.mjs` file is that it would not be automatically compatible with the rest of the backend code that uses CommonJS.
Since this is only for testing and as a standalone server, we should not need to configure the bundler to accept both. If we do in the future, we may need to research on how to do so in webpack.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "report": "istanbul report --dir ./test/coverage/total-coverage html",
     "test": "webpack --mode=production --config ./webpack.production.js && cross-env process.env.NODE_ENV=test process.env.TRAVIS_LOCAL_API=true nyc --report-dir ./test/coverage/mocha-coverage --reporter json --reporter text --reporter html mocha --exit",
     "server-sse": "node ./test/fakeSSEServer.js",
-    "server-gql": "node ./test/graphqlServer.js",
+    "server-gql": "node ./test/graphqlServer.mjs",
```
Comment (jjhuang3) [inline package.json:11]: What is an .mjs file again? Newer ES module syntax?

**Review by jjhuang3 — APPROVED** [review]
I think everything looks good. Just minor question about the mjs thing since I personally never used it.

File: package.json (hunk 1 above)
Comment (nattiechan) [reply package.json:11]:
```
MJS files mostly belong to NodeJS by OpenJS Foundation. MJS is the recommended filename extension of ECMAScript (JavaScript) modules used in NodeJS applications. NodeJS is a JavaScript runtime which allows the execution of JavaScript code outside a browser. An MJS file is essentially the same as a JS script.
```

https://www.google.com/search?q=what+is+.mjs&rlz=1C5CHFA_enCA1035CA1035&oq=what+is+.mjs&aqs=chrome..69i57j0i10i22i30j0i22i30l2j0i10i22i30j0i22i30j0i10i22i30l4.1504j0j7&sourceid=chrome&ie=UTF-8

:)

Comment (nattiechan) [reply package.json:11]: The problem is most of our backend is in CommonJS, there is no reason to change the settings in higher-levels like in `package.json` or webpack. So this is somewhat of a workaround

**Pull request merged**